libraries/db.php

```diff
@@ -0,0 +1,210 @@
+ini_set('display_startup_errors', 1);
+error_reporting(E_ALL);
+
+class dbOperation {
+    private $conn;
+    private $query;
+    private $query_result;
+    public $num_rows_result;
+    
+
+    /**
+     * To create a database connection when an object is created
+     *
+     * @access public
+     * @param void
+     * @return void 
+     */
+    public function __construct() {
+        $this->conn = mysqli_connect(SERVERNAME, USERNAME, PASSWORD, DBNAME);
+        
+        if (!$this->conn) {
+          $this->log_db_error('Connect');
+        }
+    }
+
+
+    /**
+     * To select a table from database
+     *
+     * @access public
+     * @param string $table_name Name of the table to fetch data
+     * @param array $fields_list List of columns required
+     * @param array $where_clause Array of key as column and corresrponding value as value of 
+     * that column to apply WHERE condition
+     * @param array $order_by Array containing ordering column at index 0 and ordering type at index 1
+     * @return void 
+     */
+    public function select($table_name, $fields_list=['*'], $where_clause=[], $order_by=[]) {
+        $this->query = 'SELECT '.implode(',',$fields_list)
+            . ' FROM ' . $table_name;
+
+        if ($where_clause)
+        {
+            $this->query .= ' WHERE '.implode(array_keys($where_clause)).'= \''
+                .implode(array_values($where_clause)).'\'';
+        }
+        
+        if ($order_by)
+        {
+            $order = isset($order_by[1]) ? $order_by[1] : 'ASC';
+            $this->query .= ' ORDER BY '.$order_by[0].' '.$order;
+        }     
+        $this->query_result = mysqli_query($this->conn, $this->query);
+        $this->validate_result('select');
+        $this->num_rows_result = mysqli_num_rows($this->query_result);
+    }
+
+    
+    /**
+     * To select determined attributes of all users in the database
+     *
+     * @access public
+     * @param array $where_clause Array of key as column and corresrponding value as value of that 
+     * column to apply WHERE condition
+     * @param array $order_by Array containing ordering column at index 0 and ordering type at index 1
+     * @return void
+     */
+    public function get_all_users($where_clause=[], $order_by=[]) {  
+        $this->query = "SELECT u.id,CONCAT(u.first_name,' ',u.middle_name,' ',u.last_name) as full_name,"
+            . "u.image,u.gender,u.dob,u.bio,u.preferred_comm,u.mobile,ua.type,"
+            . "CONCAT(ua.street,', ',ua.city,', ',st.name,', ',ua.zip) as address, l.email,"
+            . "CONCAT(uao.street,', ',uao.city,', ',st.name,', ',uao.zip) as office_address "
+            . "FROM `users` u "
+            . "JOIN user_address ua ON u.id=ua.user_id AND ua.type =1 "
+            . "LEFT JOIN user_address uao ON u.id=uao.user_id AND uao.type =2 "
+            . "LEFT JOIN state_table st ON ua.state= st.id OR uao.state = st.id "
+            . "JOIN `login` l ON u.id=l.user_id ";
```

Comment:
**as address** 
as is as SQL keyword it should be in CAPS. 
(make change in all places)